### PR TITLE
docs(a11y): Add a11y docs to button, button toggle, checkbox, progress spinner/ba…

### PR DESCRIPTION
**Pull request by tinayuangao**
…r, radio button and grid list

### CODE REVIEW

File: src/lib/button-toggle/button-toggle.md [hunk 1]
```diff
@@ -18,5 +18,12 @@ and `ngModel` is not supported.
 The button-toggles will present themselves as either checkboxes or radio-buttons based on the 
 presence of the `multiple` attribute. 
 
+For button toggles with icons only, Each button toggle should be marked with meaningful label by 
+`aria-label` or `aria-labelledby`.
+
+For button toggle groups, each group should be marked with meaningful label  by adding `aria-label`
```
Comment (crisbeto) [inline src/lib/button-toggle/button-toggle.md]: There's extra space between "label" and `"by".

File: src/lib/button-toggle/button-toggle.md [hunk 2]
```diff
@@ -18,5 +18,12 @@ and `ngModel` is not supported.
 The button-toggles will present themselves as either checkboxes or radio-buttons based on the 
 presence of the `multiple` attribute. 
 
+For button toggles with icons only, Each button toggle should be marked with meaningful label by
```
Comment (crisbeto) [inline src/lib/button-toggle/button-toggle.md]: "each" shouldn't be capitalized. Also here and in a few of the other files "with meaningful label" -> "with a meaningful label".

File: src/lib/progress-bar/progress-bar.md [hunk 3]
```diff
@@ -28,3 +28,6 @@ In both "indeterminate" and "query" modes, the `value` property is ignored.
 ### Theming
 The color of a progress-bar can be changed by using the `color` property. By default, progress-bars
 use the theme's primary color. This can be changed to `'accent'` or `'warn'`.  
+
+### Accessibility
+Each progress bar should be marked by  `aria-label` or `aria-labelledby` with meaningful label.
```
Comment (crisbeto) [inline src/lib/progress-bar/progress-bar.md]: "Each progress bar should be marked with an `aria-label` or ..."

File: src/lib/button-toggle/button-toggle.md (hunk 1 above)
Comment (kara) [inline src/lib/button-toggle/button-toggle.md]: Typo: missing an "a"  ("marked with a meaningful label"). Same for the usages across PR

File: src/lib/button/button.md [hunk 4]
```diff
@@ -31,3 +31,9 @@ to capitalize buttons automatically via `text-transform: uppercase`, because it
 certain locales. It is also worth noting that using ALL CAPS in the text itself causes issues for
 screen-readers, which will read the text character-by-character. We leave the decision of how to
 approach this to the consuming app.
+
+### Accessibility
+
+For buttons or links with icon only such as `md-fab`, `md-mini-fab`, `md-icon-button`, they should
```
Comment (kara) [inline src/lib/button/button.md]: Nit: might sound better to say: "Buttons or links with icons only (such as `md-fab`, `md-mini-fab`, and `md-icon-button`) should be marked with a meaningful label using `aria-label` or `aria-labelledby`.

File: src/lib/checkbox/checkbox.md [hunk 5]
```diff
@@ -26,3 +26,8 @@ remove the indeterminate state.
 ### Theming
 The color of a `<md-checkbox>` can be changed by using the `color` property. By default, checkboxes
 use the theme's accent color. This can be changed to `'primary'` or `'warn'`.  
+
+### Accessibility
+Standalone checkbox without text or labels should be marked by `aria-label` or `aria-labelledby`
```
Comment (kara) [inline src/lib/checkbox/checkbox.md]: Typo: checkbox -> checkboxes

File: src/lib/grid-list/grid-list.md [hunk 6]
```diff
@@ -41,3 +41,9 @@ however, more rows will simply be added for it the tile to fill.
 
 A header and footer can be added to an `md-grid-tile` using the `md-grid-tile-header` and
 `md-grid-tile-footer` elements respectively.
+
+### Accessibility
+Non-interactive grid list for layout purpose only should add `role=“list”` to the grid list.
```
Comment (kara) [inline src/lib/grid-list/grid-list.md]: Nit: "Non-interactive grid lists for layout purposes only should have `role="list"`."

File: src/lib/progress-spinner/progress-spinner.md [hunk 7]
```diff
@@ -21,3 +21,7 @@ In "indeterminate" mode, the `value` property is ignored.
 ### Theming
 The color of a progress-spinner can be changed by using the `color` property. By default, 
 progress-spinners use the theme's primary color. This can be changed to `'accent'` or `'warn'`.
+
+### Accessibility
+Each progress spinner should be marked with an `aria-label` or `aria-labelledby` with a meaningful
```
Comment (kara) [inline src/lib/progress-spinner/progress-spinner.md]: I think here and in progress bar, you mean "marked by aria-label or aria-labelledby with a meaningful label".

File: src/lib/button-toggle/button-toggle.md [hunk 8]
```diff
@@ -18,5 +18,12 @@ and `ngModel` is not supported.
 The button-toggles will present themselves as either checkboxes or radio-buttons based on the 
 presence of the `multiple` attribute. 
 
+For button toggles with icons only, each button toggle should be marked with a meaningful label by
```
Comment (jelbourn) [inline src/lib/button-toggle/button-toggle.md]: I would rephrase `"with icons only"` to `"containing only icons"` here and elsewhere

File: src/lib/button-toggle/button-toggle.md [hunk 9]
```diff
@@ -18,5 +18,12 @@ and `ngModel` is not supported.
 The button-toggles will present themselves as either checkboxes or radio-buttons based on the 
 presence of the `multiple` attribute. 
 
+For button toggles with icons only, each button toggle should be marked with a meaningful label by 
+`aria-label` or `aria-labelledby`.
+
+For button toggle groups, each group should be marked with a meaningful label by adding `aria-label`
```
Comment (jelbourn) [inline src/lib/button-toggle/button-toggle.md]: I would rephrase ``"marked with a meaningful label by adding `aria-label` or `aria-labelledby"`` to 
``"given a meaningful label via `aria-label` or `aria-labelledby`"`` here and elsewhere

File: src/lib/button/button.md [hunk 10]
```diff
@@ -31,3 +31,9 @@ to capitalize buttons automatically via `text-transform: uppercase`, because it
 certain locales. It is also worth noting that using ALL CAPS in the text itself causes issues for
 screen-readers, which will read the text character-by-character. We leave the decision of how to
 approach this to the consuming app.
+
+### Accessibility
+
+Buttons or links with icons only (such as `md-fab`, `md-mini-fab`, and `md-icon-button`) should
```
Comment (jelbourn) [inline src/lib/button/button.md]: I would add a short paragraph above here with something like

```markdown
Angular Material uses native `<button>` and `<a>` elements to ensure an accessible experience by default.
The `<button>` element should be used for any interaction that _performs an action on the current page_.
The `<a>` element should be used for any interaction that _navigates to another view_.
```

File: src/lib/checkbox/checkbox.md [hunk 11]
```diff
@@ -26,3 +26,8 @@ remove the indeterminate state.
 ### Theming
 The color of a `<md-checkbox>` can be changed by using the `color` property. By default, checkboxes
 use the theme's accent color. This can be changed to `'primary'` or `'warn'`.  
+
+### Accessibility
+Standalone checkboxes without text or labels should be marked by `aria-label` or `aria-labelledby`
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.md]: I would add a paragraph above here with something like
```markdown
The `<md-checkbox>` uses an internal `<input type="checkbox">` to provide an accessible experience.
This internal checkbox receives focus and is automatically labelled by the text content of the `<md-checkbox>` element.
```

File: src/lib/checkbox/checkbox.md [hunk 12]
```diff
@@ -26,3 +26,8 @@ remove the indeterminate state.
...
+with a meaningful label.
```
Comment (jelbourn) [inline src/lib/checkbox/checkbox.md]: I would omit the word "Standalone"

File: src/lib/grid-list/grid-list.md [hunk 13]
```diff
@@ -41,3 +41,9 @@ however, more rows will simply be added for it the tile to fill.
 
 A header and footer can be added to an `md-grid-tile` using the `md-grid-tile-header` and
 `md-grid-tile-footer` elements respectively.
+
+### Accessibility
+Non-interactive grid lists for layout purposes only should have `role="list".`
```
Comment (jelbourn) [inline src/lib/grid-list/grid-list.md]: I would present this as something like
```markdown
By default, the grid-list assumes that it will be used in a purely decorative fashion and thus sets no roles,
ARIA attributes, or keyboard shortcuts. This is equivalent to having a sequence of `<div>` elements on
the page. Any interactive content within the grid-list should be given an appropriate accessibility treatment
based on the specific workflow of your application.

If the grid-list is used to present a list of _non-interactive_ content items, then the grid-list element
should be given `role="list"` and each tile should be given `role="listitem"`. 
```

File: src/lib/radio/radio.md [hunk 14]
```diff
@@ -27,3 +27,7 @@ Individual radio-buttons inside of a radio-group will inherit the `name` of the
 ### Use with `@angular/forms`
 `<md-radio-group>` is compatible with `@angular/forms` and supports both `FormsModule`
 and `ReactiveFormsModule`.
+
+### Accessibility
+Radio button groups should should be marked by `aria-label` or `aria-labelledby`
```
Comment (jelbourn) [inline src/lib/radio/radio.md]: I would add a paragraph here, similar to checkbox, that mentions the internal `<input type="radio">`

**Comment by tinayuangao** [comment]
Comments addressed. Please take another look. Thanks!

File: src/lib/radio/radio.md [hunk 15]
```diff
@@ -27,3 +27,10 @@ Individual radio-buttons inside of a radio-group will inherit the `name` of the
 ### Use with `@angular/forms`
 `<md-radio-group>` is compatible with `@angular/forms` and supports both `FormsModule`
 and `ReactiveFormsModule`.
+
+### Accessibility
+The `<md-radop-button>` uses an internal `<input type="radio">` to provide an accessible experience.
```
Comment (jelbourn) [inline src/lib/radio/radio.md]: "radop" -> "radio"
